Neos.ContentRepository.Core/Classes/CommandHandler/CommandSimulator.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\CommandHandler;
+
+use Neos\ContentRepository\Core\CommandHandlingDependencies;
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\EventStore\EventNormalizer;
+use Neos\ContentRepository\Core\EventStore\EventsToPublish;
+use Neos\ContentRepository\Core\Feature\Common\RebasableToOtherWorkspaceInterface;
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphProjectionInterface;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\EventStore\Helper\InMemoryEventStore;
+use Neos\EventStore\Model\Event\SequenceNumber;
+use Neos\EventStore\Model\Events;
+use Neos\EventStore\Model\EventStream\EventStreamInterface;
+use Neos\EventStore\Model\EventStream\ExpectedVersion;
+use Neos\EventStore\Model\EventStream\VirtualStreamName;
+
+/**
+ * Implementation detail of {@see ContentRepository::handle}, when rebasing or partially publishing
+ *
+ * @internal
+ */
+final class CommandSimulator
+{
+    private bool $inSimulation = false;
+
+    /**
+     * @param ContentGraphProjectionInterface $contentRepositoryProjection
+     * @param EventNormalizer $eventNormalizer
+     * @param array<CommandHandlerInterface> $handlers
+     */
+    public function __construct(
+        private readonly CommandHandlingDependencies $commandHandlingDependencies,
+        private readonly ContentGraphProjectionInterface $contentRepositoryProjection,
+        private readonly EventNormalizer $eventNormalizer,
+        private readonly array $handlers,
+        private readonly InMemoryEventStore $inMemoryEventStore,
+        private readonly WorkspaceName $workspaceNameToSimulateIn,
+    ) {
+    }
+
+    /**
+     * @template T
+     * @param \Closure(): T $fn
+     * @return T
+     */
+    public function run(\Closure $fn): mixed
+    {
+        $this->inSimulation = true;
+        try {
+            return $this->contentRepositoryProjection->inSimulation($fn);
```

Comment:
Suggestion: Make `handle` private and change this to:
```suggestion
            return $this->contentRepositoryProjection->inSimulation($fn($this->handle(...));
```